examples/app-cpp/payload_app.cc

```diff
@@ -76,17 +76,65 @@ static void sequence_cleanup(mythreadState_t *mythread)
     return;
 }
 
-static void sequence_core(mythreadState_t *mythread)
+static void sequence_lock(mythreadState_t *mythread)
 {
-    AntarisReturnCode ret;
-    unsigned long current_time_in_ms;
-
     printf("%s: state %s, scheduled_deadline %ul\n", mythread->seq_id, mythread->state, mythread->scheduled_deadline);
 
     /* lock mutex - all cond-waits require this */
     pthread_mutex_lock(&mythread->cond_lock);
 
     printf("%s: starting sequence ...\n", mythread->seq_id);
+    return;
+}
+
+//This sequence should complete within its scheduled_deadline
+void handle_HelloWorld(mythreadState_t *mythread)
+{
+    AntarisReturnCode ret;
+
+    printf("\n Handling sequence: hello, world! \n");
+
+    // Tell PC that current sequence is done
+    CmdSequenceDoneParams sequence_done_params = {0};
+    strcpy(&sequence_done_params.sequence_id[0], HelloWorld_ID);
+    ret = api_pa_pc_sequence_done(channel, &sequence_done_params);
+
+    printf("%s: sent sequence-done notification with correlation_id %u\n", mythread->seq_id, mythread->correlation_id);
```

Comment:
I feel it is important to illustrate the importance of common/critical fields like correlation id to show how it is to be used to correlate requests with their (asynchronous) responses. Otherwise our sample code's value is limited to syntactical demonstration of API invocations.